models/aeif_cbvg_2010.cpp

```diff
@@ -0,0 +1,637 @@
+{
+  assert( this != &s ); // would be bad logical error in program
+  for ( size_t i = 0; i < STATE_VEC_SIZE; ++i )
+  {
+    y_[ i ] = s.y_[ i ];
+  }
+  r_ = s.r_;
+  clamp_r_ = s.clamp_r_;
+  return *this;
+}
+
+/* ----------------------------------------------------------------
+ * Paramater and state extractions and manipulation functions
+ * ---------------------------------------------------------------- */
+
+void
+nest::aeif_cbvg_2010::Parameters_::get( DictionaryDatum& d ) const
+{
+  def< double >( d, names::C_m, C_m );
+  def< double >( d, names::V_T_max, V_T_max );
+  def< double >( d, names::V_T_rest, V_T_rest );
+  def< double >( d, names::tau_V_T, tau_V_T );
+  def< double >( d, names::t_ref, t_ref_ );
+  def< double >( d, names::g_L, g_L );
+  def< double >( d, names::E_L, E_L );
+  def< double >( d, names::V_reset, V_reset_ );
+  def< double >( d, names::a, a );
+  def< double >( d, names::b, b );
+  def< double >( d, names::I_sp, I_sp );
+  def< double >( d, names::Delta_T, Delta_T );
+  def< double >( d, names::tau_w, tau_w );
+  def< double >( d, names::tau_z, tau_z );
+  def< double >( d, names::tau_plus, tau_plus );
+  def< double >( d, names::tau_minus, tau_minus );
+  def< double >( d, names::tau_bar_bar, tau_bar_bar );
+  def< double >( d, names::I_e, I_e );
+  def< double >( d, names::V_peak, V_peak_ );
+  def< double >( d, names::gsl_error_tol, gsl_error_tol );
+  def< double >( d, names::V_clamp, V_clamp_ );
+  def< double >( d, names::t_clamp, t_clamp_ );
+}
+
+void
+nest::aeif_cbvg_2010::Parameters_::set( const DictionaryDatum& d )
+{
+  updateValue< double >( d, names::V_T_max, V_T_max );
+  updateValue< double >( d, names::V_T_rest, V_T_rest );
+  updateValue< double >( d, names::tau_V_T, tau_V_T );
+  updateValue< double >( d, names::V_peak, V_peak_ );
+  updateValue< double >( d, names::t_ref, t_ref_ );
+  updateValue< double >( d, names::E_L, E_L );
+  updateValue< double >( d, names::V_reset, V_reset_ );
+
+  updateValue< double >( d, names::C_m, C_m );
+  updateValue< double >( d, names::g_L, g_L );
+
+  updateValue< double >( d, names::a, a );
+  updateValue< double >( d, names::b, b );
+  updateValue< double >( d, names::I_sp, I_sp );
+  updateValue< double >( d, names::Delta_T, Delta_T );
+  updateValue< double >( d, names::tau_w, tau_w );
+  updateValue< double >( d, names::tau_z, tau_z );
+  updateValue< double >( d, names::tau_plus, tau_plus );
+  updateValue< double >( d, names::tau_minus, tau_minus );
+  updateValue< double >( d, names::tau_bar_bar, tau_bar_bar );
+
+  updateValue< double >( d, names::I_e, I_e );
+
+  updateValue< double >( d, names::gsl_error_tol, gsl_error_tol );
+
+  updateValue< double >( d, names::V_clamp, V_clamp_ );
+  updateValue< double >( d, names::t_clamp, t_clamp_ );
+
+  if ( V_reset_ >= V_peak_ )
+  {
+    throw BadProperty( "Ensure that V_reset < V_peak ." );
+  }
+
+  if ( Delta_T < 0. )
+  {
+    throw BadProperty( "Delta_T must be positive." );
+  }
```

Comment:
You are right, this was poorly handled in the model. With 0ce9376 we do no longer allow `Delta_T == 0` and also added a test for numerical instability with small values of `Delta_T` similiar to the one in `aeif_psc_alpha`

Reply:
Hi there, just investing as I saw a ``Delta_T`` pass by: at some point this was explicitely added to the AdExp model to make the adaptive LIF model available for the zero value (no more exponential) notably because PyNN required it. Since the is required for the "standard" AdExp implementation, is it really a go idea to remove it from some versions?